### PR TITLE
Added enums/const for storing objects/animals/resources in SWD format

### PR DESCRIPTION
**Pull request by vader1986**
Required for random map generation.

### CODE REVIEW

File: src/enumTypes.h [hunk 1]
```diff
@@ -56,6 +56,76 @@ namespace libsiedler2
...
+    
+    /**
+     * Enumeration for object types stored in SWD. In combination with the
+     * object information it makes a real object (therefore the "info"-comments
+     * in brackets).
+     */
+    enum OBJECT_TYPES
+    {
+        OT_Empty        = 0x00,
+        OT_Stone        = 0x01, // 0x01 - 0x06 (info: 0xCC / 0xCD)
```
Comment (Flow86) [inline src/enumTypes.h]: if there are more of them, use OT_Stone_Begin OT_Stone_End or similar, so one can calculate the range

File: src/enumTypes.h [hunk 2]
```diff
@@ -56,6 +56,76 @@ namespace libsiedler2
...
+        OT_Palm         = 0xF0, // 0xF0 - 0xF7 (info: 0xC4)
+    };
+    
+    /**
+     * Enumeration for object info stored in SWD.
+     */
+    enum OBJECT_INFO
+    {
+        OI_Empty        = 0x00,
+        OI_Stone        = 0xCC, // 0xCC - 0xCD
```
Comment (Flow86) [inline src/enumTypes.h]: if there are more of them, use OI_Stone1 OI_Stone2, so we can distuingish them

File: src/enumTypes.h [hunk 3]
```diff
@@ -56,6 +56,76 @@ namespace libsiedler2
...
+        OI_TreeOrPalm   = 0xC4,
+        OI_Palm         = 0xC5,
+        OI_Headquarter  = 0x80  // used in combination with object
```
Comment (Flow86) [inline src/enumTypes.h]: name should have "Mask" in it - since its a mask for the Headquarter

File: src/enumTypes.h [hunk 4]
```diff
@@ -56,6 +56,76 @@ namespace libsiedler2
...
+                                // type 0x00-0x06 for player number
+    };
+    
+    /**
+     * Hex value to combine with texture values to enable harbor placement.
+     */
+    const unsigned char ENABLE_HARBOR = 0x40;
```
Comment (Flow86) [inline src/enumTypes.h]: same here, "HARBOR_MASK" would be better I think

**Review by Flow86 — CHANGES_REQUESTED** [review]

**Comment by vader1986** [comment]
fixed the stuff, thanks for the review

**Pull request merged**